pattern-hues.lic

```diff
@@ -54,16 +53,22 @@ class PatternHues
       /Something in the area is interfering/,
       /You try, but the dolphin choose/,
       /You should stop practicing/,
-      /You find it hard to concentrate on anything/
+      /You find it hard to concentrate on anything/,
+      /you suddenly realize that there is no custom color/,
+      /You'll interrupt your research!/
     ]
-    scripts_to_unpause = DRC.smart_pause_all
-    bput("prep cantrip pattern hues", "You are now prepared to cast the Pattern Hues cantrip.","You already have", "You should stop practicing")
+    scripts_to_unpause = DRC.safe_pause_list
+    unless scripts_to_unpause
+      DRC.message("Cannot pause, trying again shortly.")
+      return
+    end
+    DRC.bput("prep cantrip pattern hues", "You are now prepared to cast the Pattern Hues cantrip.","You already have", "You should stop practicing")
     if @settings.pattern_hues['gesture'] == "random"
-      bput("gesture #{@adj.sample} #{@color.sample}", *match_messages)
+      DRC.bput("gesture #{@adj.sample} #{@color.sample}", *match_messages)
     else
-      bput("gesture #{@settings.pattern_hues['gesture']}", *match_messages)
+      DRC.bput("gesture #{@settings.pattern_hues['gesture']}", *match_messages)
     end
-    DRC.unpause_all_list(scripts_to_unpause)
+    DRC.safe_unpause_list(scripts_to_unpause)
```

Comment:
If we are returning to the loop on a failure, we might want to move unpause down to the loop.